### DISCUSSION TITLE
Build unstable on latest `win22` tag?

### DISCUSSION THREAD

**Discussion by dandrestor** [actions/virtual-environments#8066] [General questions] [answered]
Hello everyone,

We are using the most recent `tags/win22` code to build agents in a pipeline hosted in Azure that runs every night. I noticed a lot of pipeline failures, and every time the reason appears to be problems with the build itself. To name a few:
* Chocolatey nodejs install failed with "Unable to resolve dependency: Unable to find a version of 'nodejs.install' that is compatible with 'nodejs 18.17.1 constraint: nodejs.install (= 18.17.1)'."
* Not enough disk space during CodeQL installation
* Build agent freezes while starting the installation (Azure Pipelines reports "We stopped hearing from agent xxx".)
* "404 not found" downloading Chrome WebDriver

These are all the failures I can find in our current Pipeline history, that is, since June 11th. I saw many other varied errors before, though. To be clear, these failures may happen att different code levels, since the pipeline tracks the most recent `win22` tag in the repo. Often the errors appear to be caused by issues with external packages, and I'm wondering if this kind of instability is to be expected. We have similar issues with Ubuntu as well, although less often I would say.

My questions are: 
* Is this a good way to track the latest GA image for Windows 2022?
* If yes, has anybody else been experiencing this instability?

**Comment by dandrestor** [comment]
A short clarification: we are using the latest `win22` tag. Our build script updates the repository, and then lists the tags and selects the latest tag containing the substring "win22". This is what I meant above with `windows-latest`, apologies for the misunderstanding.

An additional question: is this the right way to select the latest GA version of the Windows 2022 image?

I will update the original post also with this information.

**Reply by erik-bershel** [reply, marked as answer]
Hi @dandrestor!
Please, don't use tags to build images. You'll achieve better result with just `main` brunch.
The architecture of the project is such that when building the machine, many packages will still be the latest version, which often leads to instability of the build from the old states of the repository. We do not recommend anyone to use the old states and do not support them.

**Reply by dandrestor** [reply]
Hi Erik, thanks for responding! My current understanding is that the GA label and the support it entails applies only to the built images (available via the `-latest` label). Does Microsoft offer a similar level of support for images built by the customer directly from the repo code?

I will try switching to just `main`. Is this best practice - using `main` instead of tags - documented anywhere on the repo?

**Reply by erik-bershel** [reply]
In our [guide](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md#automated-image-generation) the first step is to clone the current state of the repo. To be honest, we don't have a direct recommendation not to use release tags - perhaps this is the reason why this practice has spread. We'll think about updating the guide.

And unfortunately we don't directly support custom builds. This is difficult to do due to the lack of repeatability. We deal with assembly issues and try to help users, but we are not responsible for the behaviour of the runners themselves, since they are guaranteed to be different from the ones we have assembled.

**Reply by dandrestor** [reply]
Thanks for the information Erik.
